### PR TITLE
Added index to Django Association model

### LINKED ISSUE #967: Django Association model missing index
The [`Association` model](https://github.com/omab/python-social-auth/blob/df6d841bbcad0a800cc0cf1c4395e0cb2531aabb/social/apps/django_app/default/models.py#L88) inherits from DjangoAssociationMixin, which [performs a lookup when logging in](https://github.com/omab/python-social-auth/blob/master/social/storage/django_orm.py#L125-L126). This lookup relies on the `server_url` and `handle` columns.

Due to the lack of indices on these columns this lookup results in a full table scan, which is non-performant for applications with large user bases.

**Recommendations**

1. Add a compound index on the `server_url` and `handle` columns of the `Association` model.
2. Update the documentation recommending that those classes inheriting `DjangoAssociationMixin` include the same compound index.

### PR DESCRIPTION
**Pull request by clintonb**
This unique_together index on the server_url and handle columns ensures lookups do not result in full table scans.

Fixes #967

### CODE REVIEW

**Comment by clintonb** [comment]
@omab please review
FYI @jibsheet

File: social/apps/django_app/default/models.py [hunk 1]
```diff
@@ -96,6 +96,9 @@ class Association(models.Model, DjangoAssociationMixin):
 
     class Meta:
         db_table = 'social_auth_association'
+        unique_together = (
```
Comment (adampalay) [inline social/apps/django_app/default/models.py]: @clintonb , does specifying two fields as `unique_together` also index them together?  I know django provides [`index_together`](https://docs.djangoproject.com/en/1.9/ref/models/options/#index-together), but wasn't sure if unique_together would provide the same functionality

Comment (clintonb) [inline social/apps/django_app/default/models.py]: Yes, they behave similarly. `unique_together` takes the extra step of enforcing uniqueness in addition to the indexing.

**Comment by rlucioni** [comment]
I think it's worth noting that running the migration in this PR against a MySQL database will incur downtime while the index is added ([reference](http://stackoverflow.com/a/14248906)).

**Comment by clintonb** [comment]
Acknowledged @rlucioni. I have also filed #970 to address the fact that this table should not store nonces forever. They should be removed once the authorization response is validated. Long-term it might make more sense to use Django's caching layer for this purpose rather than the database model.

**Comment by rlucioni** [comment]
@clintonb 👍

**Comment by rlucioni** [comment]
@omab thoughts on this?

**Comment by omab** [comment]
As per the downtime, handling that is out of the scope of a simple lib, in fact the possible solutions mention in the stackoverflow question are mostly architectural. Still, thanks for the notice.

About the nounces, I agree that they shouldn't reside in the DB and I've been planning on removing that table for a while, probably just use the framework of choise session mechanism.

**Comment by omab** [comment]
Thanks!

**Comment by rlucioni** [comment]
@omab would it be possible to publish v0.2.20 so that the community can take advantage of this change?

**Comment by omab** [comment]
@rlucioni, `v0.2.20` released to pypi.

**Comment by rlucioni** [comment]
Thank you @omab!

**Pull request merged**